docs/source/dev/tests.md

```diff
@@ -0,0 +1,92 @@
+# Automated Testing
+
+Testing is a crucial aspect of software development that ensures the reliability, quality, and performance of an application. Additionally, testing facilitates better code design and architecture by encouraging modular and decoupled code. Hence, this document will focus on the different testing frameworks used in this project, explain how to correctly use them and how to contribute to the testing environment.
+
+The tests in this project include two different frameworks, namely `pytest` and `cypress`. The focus of both testing frameworks is slightly different:
```

Comment:
In general it is always a good idea to add links...

```suggestion
The tests in this project include two different frameworks, namely [`pytest`]8https://pytest.org/) and [`cypress`](https://www.cypress.io/). The focus of both testing frameworks is slightly different:
```

Reply:
That's a good point, I added links, where necessary and marked the pr as ready for review, thanks!